tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,354 @@
+            continue
+
+        is_running = is_container_running(duthost, container_name)
+        if not is_running:
+            pytest.fail("{} container is not running. Exiting...".format(container_name))
+
+        logger.info("Start testing the {} container...".format(container_name))
+
+        need_restore_state = False
+        if container_autorestart_states[container_name] == "disabled":
+            logger.info("Change auto-restart state of {} container to be 'enabled'".format(container_name))
+            duthost.shell("config container feature autorestart {} enabled".format(container_name))
+            need_restore_state = True
+
+        # Currently we select 'rsyslogd' as non-critical processes for testing based on
+        # the assumption that every container has an 'rsyslogd' process running and it is not
+        # considered to be a critical process
+        program_status, program_pid = get_program_info(duthost, container_name, "rsyslogd")
+        verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd",
+                                                        program_status, program_pid)
+
+        critical_group_list, critical_process_list = get_critical_group_and_process_list(duthost, container_name)
+        for critical_process in critical_process_list:
+            program_status, program_pid = get_program_info(duthost, container_name, critical_process)
+            verify_autorestart_with_critical_process(duthost, container_name, critical_process,
+                                                     program_status, program_pid)
+            # Sleep 20 seconds in order to let the processes come into live after container is restarted.
+            # We will uncomment the following line once the "extended" mode is added
+            # time.sleep(20)
+            # We are currently only testing one critical process, that is why we use 'break'. Once
+            # we add the "extended" mode, we will remove this statement
+            break
+
+        for critical_group in critical_group_list:
+            group_program_info = get_group_program_info(duthost, container_name, critical_group)
+            for program_name in group_program_info:
+                verify_autorestart_with_critical_process(duthost, container_name, program_name,
+                                                         group_program_info[program_name][0],
+                                                         group_program_info[program_name][1])
+                # We are currently only testing one critical program for each critical group, which is
+                # why we use 'break' statement. Once we add the "extended" mode, we will remove this
+                # statement
+                break
+
+        # After these three containers are restarted, we need wait to give their dependent containers
+        # a chance to restart
+        if container_name in ["syncd", "swss", "database"]:
```

Comment:
For now, let's omit the database container from our testing, as autorestart is not fully support for that container, and we are disabling it by default.

Reply:
Updated!